editor/src/components/canvas/gap-utils.ts

```diff
@@ -153,6 +171,157 @@ export function gapControlBoundsFromMetadata(
   )
 }
 
+export function gridGapControlBoundsFromMetadata(
+  elementMetadata: ElementInstanceMetadataMap,
+  parentPath: ElementPath,
+  gridRowColumnInfo: GridData,
+  gapValues: { row: CSSNumber; column: CSSNumber },
+  scale: number,
+): {
+  gaps: Array<{
+    bounds: CanvasRectangle
+    gapId: string
+    gap: CSSNumber
+    axis: Axis
+  }>
+  rows: number
+  columns: number
+  cellBounds: CanvasRectangle
+  gapValues: { row: CSSNumber; column: CSSNumber }
+  gridTemplateRows: string
+  gridTemplateColumns: string
+} {
+  const parentGridPlaceholderId = `grid-${EP.toString(parentPath)}`
+  const parentGrid = document.getElementById(parentGridPlaceholderId)
```

Comment:
as discussed on Discord, let's not use the DOM query to get the parent grid. We can do this based on the parent path, let's use that to stay consistent

Reply:
this is not getting the actual element here, but actually the grid placeholder - the one that is drawn in the grid controls, so no path for it.
(also mentioned it in my reply on Discord)

Reply:
Makes sense, the main problem with this is that this is a huge side effect, so if we're going to start using this from multiple places we should extract this into a function that has a big exclamation mark attached
otherwise I'm afraid that this will become a little gotcha that we'll have to keep in mind all the time when we're working on the grid code